### PR TITLE
ENH: External Stylesheet Support

### PR DESCRIPTION
**Pull request by hhslepicka**
This PR supersedes #361.
Couple of changes since that PR:
- Rebased to latest master
- Fixed default stylesheet to behave like PyDM was before.
- Fixed a couple of issues with the widgets
- Fixed old bug in which a widget without channel specified was flagged as `DISCONNECTED`.
- Another minor fixes and tests adjustments.

This one is good for a final review and merge if agreed.
The documentation section will come in a following PR soon.

### CODE REVIEW

File: pydm/widgets/tab_bar.py [hunk 1]
```diff
@@ -10,12 +10,16 @@ class PyDMTabBar(QTabBar, PyDMWidget):
...
         self.tab_channels = {}
+        self.tab_connection_status = {}
+        self.tab_alarm_severity = {}
         self._channels = None
-        self.alarm_icons = (IconFont().icon('circle', color=self.qcolor_for_alarm(self.ALARM_NONE, alarm_type=self.ALARM_INDICATOR)), 
-                        IconFont().icon('circle', color=self.qcolor_for_alarm(self.ALARM_MINOR, alarm_type=self.ALARM_INDICATOR)),
-                        IconFont().icon('exclamation-circle', color=self.qcolor_for_alarm(self.ALARM_MAJOR, alarm_type=self.ALARM_INDICATOR)),
-                        IconFont().icon('question-circle', color=self.qcolor_for_alarm(self.ALARM_INVALID, alarm_type=self.ALARM_INDICATOR)),
-                        IconFont().icon('times-circle', color=self.qcolor_for_alarm(self.ALARM_DISCONNECTED, alarm_type=self.ALARM_INDICATOR)))
+        self._no_alarm_icon_color = QColor(0,220,0)
```
Comment (hmbui) [inline pydm/widgets/tab_bar.py]: Need spacing after the commas.

File: pydm/widgets/tab_bar.py [hunk 2]
```diff
@@ -10,12 +10,16 @@ class PyDMTabBar(QTabBar, PyDMWidget):
...
+        self._minor_alarm_icon_color = QColor(220, 220, 0)
+        self._major_alarm_icon_color = QColor(255, 0, 0)
+        self._invalid_alarm_icon_color = QColor(220, 0, 220)
+        self._disconnected_alarm_icon_color = QColor(255,255,255)
```
Comment (hmbui) [inline pydm/widgets/tab_bar.py]: Same comment as above.

**Comment by hhslepicka** [comment]
Thank you @hmbui for your review. I just changed the code to address your comments.

File: pydm/widgets/base.py [hunk 3]
```diff
@@ -272,6 +165,8 @@ def connection_changed(self, connected):
         self.check_enable_state()
         if not connected:
             self.alarmSeverityChanged(self.ALARM_DISCONNECTED)
+        else:
+            self.alarmSeverityChanged(self.ALARM_NONE)
```
[the remark this answers is not in the document]
Comment (hhslepicka) [reply pydm/widgets/base.py]: That would only be possible if a plugin would send the alarm status before sending the connection status... this is something that we should enforce the plugins to do properly...
In the cases in which I tested all the widgets behaved nicely...
I will do one more test and mark this as solved if it pass... which is:
- Start an IOC...
- Put it in an alarm state...
- Drop my laptop connection...
- Reconnect my laptop...
- Check if the widget will show the proper alarm state.

File: pydm/widgets/base.py [hunk 4]
```diff
@@ -525,9 +431,10 @@ def alarmSensitiveContent(self, checked):
             alarm severity changes.
         """
         self._alarm_sensitive_content = checked
-        self._alarm_flags = (self.ALARM_CONTENT * self._alarm_sensitive_content) | (self.ALARM_BORDER * self._alarm_sensitive_border)
-        if is_pydm_app():
-            self.alarm_severity_changed(self._alarm_state)
+        self.alarm_severity_changed(self._alarm_state)
+        self.style().unpolish(self)
```
[the remark this answers is not in the document]
Comment (hhslepicka) [reply pydm/widgets/base.py]: That is right... this is not needed.

File: pydm/widgets/base.py [hunk 5]
```diff
@@ -555,9 +462,10 @@ def alarmSensitiveBorder(self, checked):
             alarm severity changes.
         """
         self._alarm_sensitive_border = checked
-        self._alarm_flags = (self.ALARM_CONTENT * self._alarm_sensitive_content) | (self.ALARM_BORDER * self._alarm_sensitive_border)
-        if is_pydm_app():
-            self.alarm_severity_changed(self._alarm_state)
+        self.alarm_severity_changed(self._alarm_state)
+        self.style().unpolish(self)
```
[the remark this answers is not in the document]
Comment (hhslepicka) [reply pydm/widgets/base.py]: That is right... this is not needed.

**Comment by hhslepicka** [comment]
Rebased and force pushed to solve conflicts.
@mattgibbs it is ready for another quick review before merging.

**Pull request merged**